src/utils/spline_builder_deriv_field_2d.hpp

```diff
@@ -0,0 +1,266 @@
+     * @param[out] function Field with layout_right where we copy the function values.
+     * @param[in] function_and_derivs DerivField from where the function values are copied.
+     */
+    void fill_in_function(FunctField function, DerivFieldType function_and_derivs) const
+    {
+        // Fill the field with correct layout.
+        ddc::parallel_for_each(
+                ExecSpace(),
+                get_idx_range(function_and_derivs),
+                KOKKOS_LAMBDA(Idx<Grid1, Grid2> const idx) {
+                    function(idx) = function_and_derivs(idx);
+                });
+    }
+
+    /**
+     *  @brief Fill in the deriv1 field with the derivatives along the first dimension stored 
+     * in the function_and_derivs. 
+     * @param[out] deriv1 Field with layout_right where we copy the derivatives.
+     * @param[in] function_and_derivs DerivField from where the derivatives are copied.
+     * @param[in] is_min Boolean which helps to determine which bound (mon/max) we select for the derivative field. 
+     */
+    void fill_in_deriv1(Deriv1Field deriv1, DerivFieldType function_and_derivs, bool const is_min)
+            const
+    {
+        IdxRangeSlice<Grid1> idx_range_slice_1
+                = function_and_derivs.template idx_range_for_deriv<Grid1>();
+        Idx<Grid1> idx_slice = is_min ? idx_range_slice_1.front() : idx_range_slice_1.back();
+
+        // Fill the field with correct layout.
+        ddc::parallel_for_each(
+                ExecSpace(),
+                get_idx_range(deriv1),
+                KOKKOS_LAMBDA(Idx<Deriv1, Grid2> const idx) {
+                    Idx<Deriv1> idx_d1(idx);
+                    Idx<Grid2> idx_2(idx);
+                    deriv1(idx) = function_and_derivs(idx_d1, idx_slice, idx_2);
+                });
+    }
+
+    /**
+     *  @brief Fill in the deriv2 field with the derivatives along the second dimension stored 
+     * in the function_and_derivs. 
+     * @param[out] deriv2 Field with layout_right where we copy the derivatives.
+     * @param[in] function_and_derivs DerivField from where the derivatives are copied.
+     * @param[in] is_min Boolean which helps to determine which bound (mon/max) we select for the derivative field. 
+     */
+    void fill_in_deriv2(Deriv2Field deriv2, DerivFieldType function_and_derivs, bool const is_min)
+            const
+    {
+        IdxRangeSlice<Grid2> idx_range_slice_2
+                = function_and_derivs.template idx_range_for_deriv<Grid2>();
+        Idx<Grid2> idx_slice = is_min ? idx_range_slice_2.front() : idx_range_slice_2.back();
+
+        // Fill the field with correct layout.
+        ddc::parallel_for_each(
+                ExecSpace(),
+                get_idx_range(deriv2),
+                KOKKOS_LAMBDA(Idx<Grid1, Deriv2> const idx) {
+                    Idx<Grid1> idx_1(idx);
+                    Idx<Deriv2> idx_d2(idx);
+                    deriv2(idx) = function_and_derivs(idx_d2, idx_slice, idx_1);
```

Comment:
It should be possible to just do
```suggestion
                    deriv2(idx) = function_and_derivs(idx, idx_slice);
```